tsdb/db.go

```diff
@@ -844,6 +844,10 @@ func (db *DB) Appender(ctx context.Context) storage.Appender {
 }
 
 func (db *DB) ApplyConfig(conf *config.Config) error {
+	if db.opts.AllowOverlappingBlocks && !conf.StorageConfig.TSDB.Retention.AllowOverlappingBlocks {
+		return errors.Errorf("restart prometheus to disable overlapping blocks")
+	}
+
```

Comment:
We should also be applying these config to the DB for these changes to be reflected.

You will have to make these variables atomic and update them accordingly here:
```
db.opts.MaxBytes
db.opts.RetentionDuration
db.opts.AllowOverlappingBlocks
```